src/core/thread/topology.hpp

```diff
@@ -351,6 +369,10 @@ class Neighbor
     bool            mDataRequest : 1; ///< Indicates whether or not a Data Poll was received
     uint8_t         mLinkFailures;    ///< Consecutive link failure count
     LinkQualityInfo mLinkInfo;        ///< Link quality info (contains average RSS, link margin and link quality)
+
+#if OPENTHREAD_CONFIG_ENABLE_TIME_SYNC
+    bool mTimeSyncEnabled : 1; ///< Indicates whether or not time sync feature is enabled.
```

Comment:
Adding the bool will increase the  struct size by 4 bytes (to keep the struct aligned). 
I don't think `mlLinkFailures` really requires 8 bit. @jwhui what do you think?
How about if we change `mLinkFailures` to be 7 bits and then declare the `bool mTimeSyncEnabled` after it?